sstables/sstable_mutation_reader.hh

```diff
@@ -109,16 +109,16 @@ position_in_partition_view get_slice_lower_bound(const schema& s, const query::p
 // The amount of this excessive read is controlled by read ahead
 // heuristics which learn from the usefulness of previous read aheads.
 template <typename DataConsumeRowsContext>
-inline std::unique_ptr<DataConsumeRowsContext> data_consume_rows(const schema& s, shared_sstable sst, typename DataConsumeRowsContext::consumer& consumer,
+inline future<std::unique_ptr<DataConsumeRowsContext>> data_consume_rows(const schema& s, shared_sstable sst, typename DataConsumeRowsContext::consumer& consumer,
```

Comment:
Please, coroutinize data_consume_rows() as separate pre-patch (always co_returning the result without co_await-ing anything) to reduce churn here